ThunkLibs/libX11/libX11_Host.cpp

```diff
@@ -25,103 +25,301 @@ desc: Handles callbacks and varargs
 #include <utility>
 
 #include "ldr_ptrs.inl"
+#ifdef _M_ARM_64
+__attribute__((naked))
+void* InternalVariadic(void* Ptr, uint64_t a_0, size_t count, void** list) {
+  // Arguments 0-7 is in registers
+  // 8+ stored on to stack
+  // 0 need to be on the end of the list
```

Comment:
Does this refer to the input list or to the output arguments? Wording is unclear; I thought it's the input list at first, but looks like you mean the outputs.

Reply:
Updated comment to be more clear.
```
// nullptr needs to be at the end of the list of generated stack items for this API.
// We will always generate a variadic frame of `count` objects + 1 for nullptr.
```

Reply:
Unfortunately I find this even more confusing, now that this is just a blob of inline-assembly without the context that was previously provided by `void* InternalVariadic(void* Ptr, uint64_t a_0, size_t count, void** list)`.

I'll try to put this into more concrete terms: "This API" isn't unclear about whether it refers to the called function or the blob provided by VariadicImplementation.inl (which isn't really an "interface" at this point). This in turn allows for reading "generated stack items" in two different ways (the array generated on the guest-side or the stack contents generated here?). At the end of the comment I'm left thinking "Did you just describe a requirement for using this blob or an implementation detail?".

Hope that helps explain where my confusion is coming from. I can see us using this feature more widely in other libraries in the future, so having clear phrasing would be useful to have.

Reply:
Updated the comment more to explain that the functions we are calling are expecting nullptr to be at the end of the stack frame, not the stack frame coming in to the function.

Reply:
Much better with the updated comment and with the old context around again. Thanks!